### ISSUE TITLE
Add sql timeout to custom timing

### ISSUE THREAD

**Issue by NickLarsen** [MiniProfiler/dotnet#254]
TL;DR;

I am using attempting to use miniprofiler in a console application as a means of collecting query information for all of our sql queries.  Our specific use case is to identify queries that are running close to the timeouts we have specified.  There are hundreds of dynamic queries in this application, so lining them up by hand is not super realistic long term.  I'm trying to find a way to augment miniprofiler custom timings with the timeout information.

---
As our data has grown (tens of millions of rows per day for some typical data sets), some of our queries are starting to timeout.  We'd like to record the queries, their runtimes and their timeouts in a queryable manner.  Setting everything up to use miniprofiler was a breeze with the exception that it does not support recording the timeout out of the box.

First question is, is there some way to make this work without requiring library updates that does not involve writing my own profiler implementation?

If yes, HOORAY! please share.

If no, second question is, is this something that is reasonable to add?

---
I was unable to find a simple hook in miniprofiler to solve this problem and 3 options seemed like they could work.

1. add a `Timeout` field to the `CustomTiming` class
2. add a more generic `AdditionalData` collection to the `CustomTiming` class
3. write my own profiler implementation that supports option 1

I'm really trying to avoid option 3 if possible since miniprofiler already handles 99% of what I need.

**Comment by NickCraver** [comment]
Timeout is *generally* useful for all, so I wouldn't go adding that route, for example there's a DNS timeout, connection timeout, and execution timeout - which do we mean? I get that you're after execution timeouts here, so let's see how we can address that without adding weight for all.

How are you spinning up `ProfiledDbConnection`? While it generally takes a `MiniProfiler`, for cases like this it's intentionally a much more constrained interface than that: `IDbProfiler`. There are only 4 methods on this interface, so you could create a version of this that (proxying to `Current.MiniProfiler` itself) adds a timeout to the message, or abuse the `Category` string in ways we should never speak of, or something along those lines.

Does something like that cover what you need without a lot of work?

**Comment by NickCraver** [comment]
Closing this out to cleanup - let me know if the above doesn't work!
